include/runtime/header.h

```diff
@@ -207,6 +210,14 @@ void printSortedConfigurationToFile(
     FILE *file, block *subject, char const *sort);
 void printConfigurationInternal(
     writer *file, block *subject, const char *sort, bool, void *);
+
+// This function injects its argument into KItem before printing, using the sort
+// argument as the source sort. Doing so allows the term to be pretty-printed
+// using the existing recursion scheme code (and without manually inspecting the
+// sort to see what printer we need to call if the term isn't an entire
+// configuration).
+string *debug_print_term(block *subject, char const *sort);
```

Comment:
Is there a reason besides convenience we did it this way? I would tend to think we should try to handle it in a more principled manner if we can, because we debug k programs often enough in contexts where we want to see the exact kore that the term represents, and this technically alters it, which will be visible to the user if they disable the pretty printer for kore.

Reply:
Yes, I agree that a more principled way would be preferable. The problem is roughly as follows:

The [existing configuration printer entrypoint](https://github.com/runtimeverification/llvm-backend/blob/5bcdedd51a7874afcb19af5340b4af85d0c37830/runtime/util/ConfigurationPrinter.cpp#L114-L116) assumes that its argument is a block (and we can therefore inspect the pointer bits). The generated-code recursion scheme handles calling the `print$SORT` family of functions safely once we know the sort of a term _below the top level_.

We could dispatch over the known sort of the term roughly like this to get similar behaviour:
```c++
if (sort_str == "SortInt{}") printInt(...)
else if (sort_str == "SortBool{}") printBool(...)
...
```

The issue I came to is that `printList` (c.f. sets, maps) require that the collection symbols (unit, element, concat) are known if you have a term of `SortList{}`. We could hard-code `Lbl'Unds'List'Unds` etc., but I wasn't sure if this is potentially unsound given that the recursion code has to[ look up the symbols in the definition](https://github.com/runtimeverification/llvm-backend/blob/79c01a60fa1bf3219a189e23fe3003e7ed7313c4/lib/codegen/EmitConfigParser.cpp#L887-L890). I note that this is actually what the GDB code does, but in the interests of keeping the code small I opted for the "inject into KItem" possible unsoundness rather than the "guess the collection symbols" one.

Let me know your thoughts here - happy to change the approach to whatever is the most sensible.

Reply:
Hmmm. I guess this is a reasonable tradeoff. Let's go ahead and keep this as is for now.